### PR TITLE
feat(shared-addresses): add loading state for checking permissions

### LINKED ISSUE #11893: Add a basic Loading indicator in the Choose shared addresses panel
### Description

Recalculating permissions can take a few seconds when switching around accounts, so we need a way to tell the user that something is happening.

For now, add a very basic indicator that we are calculating.

### Acceptance Criteria

- There is a loading indicator when a shared address is changed in the Choose shared addresses panel

### PR DESCRIPTION
**Pull request by jrainville**
Fixes #11893

Joining:
[joining-loading.webm](https://github.com/status-im/status-desktop/assets/11926403/1720ac76-f449-4246-bddb-b27f081e1b2e)

Editing:
[edit-addr-loading.webm](https://github.com/status-im/status-desktop/assets/11926403/fb56da0d-a81f-4366-8f37-f14713417567)

### CODE REVIEW

**Review by mprakhov — APPROVED** [review]
LGTM

File: ui/app/AppLayouts/Communities/panels/RequirementsCheckPendingLoader.qml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+import QtQuick.Layouts 1.15
+import QtGraphicalEffects 1.15
+
+import StatusQ.Core 0.1
+import StatusQ.Core.Theme 0.1
+
+StatusBaseText {
+    id: requirementsCheckPendingText
+
+    Layout.alignment: Qt.AlignHCenter
```
Comment (caybro) [inline ui/app/AppLayouts/Communities/panels/RequirementsCheckPendingLoader.qml:11]: This kinda assumes the component will be used inside a Layout

File: ui/app/AppLayouts/Communities/panels/SharedAddressesPanel.qml [hunk 2]
```diff
@@ -174,6 +176,15 @@ Control {
             }
         }
 
+        RequirementsCheckPendingLoader {
+            visible: root.requirementsCheckPending
+        }
+
+        Item {
+            width: parent.width
```
Comment (caybro) [inline ui/app/AppLayouts/Communities/panels/SharedAddressesPanel.qml:184]:
```suggestion
            Layout.fillWidth: true
```

Comment (caybro) [reply ui/app/AppLayouts/Communities/panels/SharedAddressesPanel.qml:184]: Here you are inside a Layout

File: ui/app/AppLayouts/Communities/panels/RequirementsCheckPendingLoader.qml [hunk 3]
```diff
@@ -0,0 +1,25 @@
+import QtQuick 2.15
+import QtQuick.Layouts 1.15
+import QtGraphicalEffects 1.15
```
Comment (caybro) [inline ui/app/AppLayouts/Communities/panels/RequirementsCheckPendingLoader.qml:3]: Don't need this import

File: ui/app/AppLayouts/Communities/panels/RequirementsCheckPendingLoader.qml [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+import StatusQ.Core 0.1
+import StatusQ.Core.Theme 0.1
+
+StatusBaseText {
+    id: requirementsCheckPendingText
```
Comment (caybro) [inline ui/app/AppLayouts/Communities/panels/RequirementsCheckPendingLoader.qml:9]:
```suggestion
    id: root
```

File: ui/app/AppLayouts/Communities/panels/RequirementsCheckPendingLoader.qml (hunk 1 above)
Comment (caybro) [reply ui/app/AppLayouts/Communities/panels/RequirementsCheckPendingLoader.qml:11]: But it's fine, it this was to be used inside e.g. a `Column`, it's a noop

File: ui/app/AppLayouts/Communities/panels/SharedAddressesPanel.qml [hunk 5]
```diff
@@ -174,6 +176,15 @@ Control {
...
+            height: Style.current.padding
```
Comment (caybro) [inline ui/app/AppLayouts/Communities/panels/SharedAddressesPanel.qml:185]:
```suggestion
            Layout.preferredHeight: Style.current.padding
```

**Review by caybro — COMMENTED** [review]
Looks good in general, just some minor things inline

**Review by caybro — APPROVED** [review]
LGTM

**Pull request merged**